src/server/context_test.ts

```diff
@@ -21,11 +21,20 @@ Deno.test("pathToPattern", async (t) => {
 
   await t.step("parses multiple params in same part", () => {
     assertEquals(pathToPattern("foo/[mod]@[version]"), "/foo/:mod@:version");
-
     assertEquals(pathToPattern("foo/[bar].json"), "/foo/:bar.json");
     assertEquals(pathToPattern("foo/foo[bar]"), "/foo/foo:bar");
   });
 
+  await t.step("parses optional params", () => {
+    assertEquals(pathToPattern("foo/[[name]]"), "/foo{/:name}?");
+    assertEquals(pathToPattern("foo/[name]/[[bob]]"), "/foo/:name{/:bob}?");
+    assertEquals(pathToPattern("foo/[[name]]/bar"), "/foo{/:name}?/bar");
+    assertEquals(
+      pathToPattern("foo/[[name]]/bar/[[bob]]"),
+      "/foo{/:name}?/bar{/:bob}?",
+    );
+  });
```

Comment:
Let's add one test case for invalid usage like:

```sh
foo/[[name]]-bar/baz
```

Currently, that returns this path `/foo{/:name}?-bar/baz` which would match `foo-bar/baz`. I think adding a check if the next character after the second closing `]` is either the end of the string, or a `/` character should suffice.

Reply:
thanks for writing down the tests cases, this made it incredible easy to add!